### ISSUE TITLE
✨Replace graphql endpoints deprecated by Roles

### ISSUE THREAD

**Issue by tristan-orourke** [GCTC-NTGC/gc-digital-talent#5441]
blocked by #5438 

## ✨ Feature

In #5438, we marked as deprecated any existing graphql endpoints which cannot by guarded by the new permissions. For those cases, we want to write new endpoints which can be guarded appropriately and still allow admins to do their work.

## 🕵️ Details

These deprecated endpoints are probably a problem of overly expansive scope - `any` scoped queries that should be replaced by `team` scoped ones. In the new endpoint, team name or endpoint may need to be included as an argument. Or perhaps, something like a Pool id would be appropriate, and the permission can check the team associated with that resource indirectly.

## 🎨 Design File

Include a link to the design file (if it exists).

## 📸 Screenshot

Add a screenshot of the new feature (if possible).

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] New graphql endpoints to replace permission-deprecated ones
- [ ] New endpoints are protected by policy methods
- [ ] Use of deprecated queries on frontend is replaced with new queries ?
- [ ] Deprecated queries are deleted ?

**Comment by tristan-orourke** [comment]
There's actually only one endpoint which needs to be removed, so I'm closing this in favour of the more specific #6364
